apps/src/lab2/views/components/versionHistory/VersionHistoryDropdown.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+
+import {Button, buttonColors} from '@cdo/apps/componentLibrary/button';
+import Lab2Registry from '@cdo/apps/lab2/Lab2Registry';
+import {
+  setProjectSource,
+  setAndSaveProjectSource,
+} from '@cdo/apps/lab2/redux/lab2ProjectRedux';
+import {ProjectSources, ProjectVersion} from '@cdo/apps/lab2/types';
+import {commonI18n} from '@cdo/apps/types/locale';
+import {useAppDispatch} from '@cdo/apps/util/reduxHooks';
+
+import moduleStyles from './version-history.module.scss';
+
+interface VersionHistoryDropdownProps {
+  versionList: ProjectVersion[];
+  updatedSourceCallback?: (source: ProjectSources) => void;
+  startSource: ProjectSources;
+  closeDropdown: () => void;
+}
+
+/**
+ * Dropdown that displays a list of versions for the current project.
+ * Each version has a "restore" button that will restore the project to that version.
+ * There is also a "start over" button that will restore the project to the start source.
+ */
+const VersionHistoryDropdown: React.FunctionComponent<
+  VersionHistoryDropdownProps
+> = ({versionList, updatedSourceCallback, startSource, closeDropdown}) => {
+  const dispatch = useAppDispatch();
+  const restoreVersion = (version: ProjectVersion) => {
+    const projectManager = Lab2Registry.getInstance().getProjectManager();
+    if (projectManager) {
+      projectManager.restore(version.versionId).then(sources => {
+        if (sources) {
+          dispatch(setProjectSource(sources));
+          if (updatedSourceCallback) {
+            updatedSourceCallback(sources);
+          }
+        } else {
+          // TODO: handle no source
+          console.error('no source found!');
+        }
+        closeDropdown();
+      });
+    }
+  };
+
+  const startOver = () => {
```

Comment:
Should this use the existing Start Over dialog?

Reply:
Good idea, I'll add that to the follow up ticket.